src/emqtt.erl

```diff
@@ -1223,22 +1403,26 @@ is_inflight_full(#state{max_inflight = MaxLimit, inflight = Inflight}) ->
 delete_inflight(?PUBACK_PACKET(PacketId, ReasonCode, Properties),
                 State = #state{inflight = Inflight}) ->
 	case maps:find(PacketId, Inflight) of
-        {ok, {publish, #mqtt_msg{packet_id = PacketId}, _Ts}} ->
-            ok = eval_msg_handler(State, puback, #{packet_id   => PacketId,
-                                                   reason_code => ReasonCode,
-                                                   properties  => Properties}),
+        {ok, ?INFLIGHT_PUBLISH(
+                _Msg = #mqtt_msg{packet_id = PacketId},
+                _SentAt, _ExpireAt, Callback
+               )} ->
+
+            eval_callback_handler(
+              {ok, #{packet_id => PacketId,
+                     reason_code => ReasonCode,
+                     reason_code_name => reason_code_name(ReasonCode),
+                     properties => Properties
+                    }}, Callback),
             State#state{inflight = maps:remove(PacketId, Inflight)};
```

Comment:
add delete_inflight API? so it's consistent to update_inflight.

Reply:
Or maybe make `update_inflight` act directly on the state, as this `delete_inflight` does?